peps/pep-0729.rst

```diff
@@ -0,0 +1,414 @@
+force them to implement particular features or make behavior changes. Type
+checkers are incentivized to follow the specification set out by the Council
+because it allows them to take advantage of shared resources, such as
+libraries that expose typing information that follows the specification,
+the stub files in typeshed, the ``typing`` standard library module, and
+user documentation that covers the standard type system.
+Type checkers are free to extend the type system or deviate from the
+specification, but they should document such differences clearly.
+
+The fact that type checkers need to implement any decisions made by the
+Typing Council acts as a useful brake on the Council, ensuring that its
+decisions are conservative and well-considered. Individual type checkers
+remain free to innovate as they see fit, and successful innovations can
+be incorporated into the standard type system.
+
+Projects
+--------
+
+Here are some efforts a Typing Council would be responsible for.
+
+Conformance test suite
+^^^^^^^^^^^^^^^^^^^^^^
+
+A conformance test suite would provide machine checkable documentation for how
+type checkers should check Python code, accompanied by the results of major type
+checker implementations on the test suite. A rough sketch for what this could
+look like was `created by Shantanu <https://github.com/hauntsaninja/type_checker_consistency>`_.
+
+This would contain prescriptive tests from behavior prescribed by previous PEPs
+and descriptive tests that let us document behavior of existing implementations
+in areas that are not prescribed by any standard. These descriptions would be
+useful to inform efforts below and to identify areas of focus for
+standardization.
+
+Specification for the type system
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A specification would initially be created by stitching together the
+specification sections from the existing PEPs, and then gradually improved to
+clarify points of confusion and cover more areas. A draft of such a
+stitched-together spec was `created by Jelle <https://github.com/JelleZijlstra/typing-spec>`_.
+
+The specification has a few audiences:
+
+* For type checkers, it provides a description of how an idealized type checker
+  should behave. Individual type checkers have different goals and technical
+  constraints and they are free to deviate from the spec if they do not have the
+  resources to fully implement it or if they believe a different behavior better
+  serves their users. However, they should document such deviations from the
+  spec.
+* For projects such as typeshed, or libraries that want to be compatible with
+  multiple type checkers, it provides a set of rules that they can follow to
+  make their code understood by type checkers.
+* For people who want to propose changes to the type system, it provides a
+  foundation for any new proposals.
+
+Notably, the specification is not aimed at end users of typing. End users
```

Comment:
Hm. But library maintainers mentioned above are also just end users.

Reply:
Rewording this to say it's about users who don't need to worry about multiple type checkers (which is likely most normal developers).